src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -259,10 +259,16 @@ private static (string[], int[], bool[], TFShape[], TFDataType[]) GetInputMetaDa
 
                     tfShapes[i] = graph.GetTensorShape(tfoutput);
                     var type = inputSchema.GetColumnType(inputColIndices[i]);
-                    var shape = tfShapes[i].ToIntArray();
-                    int valCount = tfShapes[i].ToIntArray().Skip(tfShapes[i][0] == -1 ? BatchSize : 0).Aggregate((x, y) => x * y);
-                    if (type.ValueCount != valCount)
-                        throw Contracts.Except($"The size of model input '{colNames[i]}' does not match its size in the input data.");
+                    var shape = tfShapes[i].ToIntArray().Skip(tfShapes[i][0] == -1 ? BatchSize : 0);
+                    if (type.AsVector.DimCount == 1)
+                    {
+                        int valCount = shape.Aggregate((x, y) => x * y);
```

Comment:
So if data is 1d we will verify  with the product of dimensions in the model.  But if data passed in is multi-dimensional then we verify if each of the individual dimensions match.  Is that the intent of this change..

do you think it might be useful to also display the shapes that mismatch ?